ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -113,53 +82,58 @@ message CreatedEvent {
+// Records that a contract has been transferred-out, and it becomes unusable on the source domain
+message TransferredOutEvent {
 
-  // The interface implemented by the matched event.
+  // The ID of the transfer out event. This needs to be used as an input for a transfer-in TransferCommand.
+  // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
-  Identifier interface_id = 1;
+  string transfer_out_id = 1;
 
-  // Whether the view was successfully computed, and if not,
-  // the reason for the error. The error is reported using the same rules
-  // for error codes and messages as the errors returned for API requests.
+  // The ID of the created contract.
+  // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
-  google.rpc.Status view_status = 2;
+  string contract_id = 2;
 
-  // The value of the interface's view method on this event.
-  // Set if it was requested in the ``InterfaceFilter`` and it could be
-  // sucessfully computed.
-  // Optional
-  Record view_value = 3;
+  // The ID of the source domain
+  // Must be a valid domain ID
+  // Required
+  string source = 3;
+
+  // The ID of the target domain
+  // Must be a valid domain ID
+  // Required
+  string target = 4;
 
+  // Transfer-in exclusivity
+  // Before this time, only the initiator of the transfer-out can initiate the transfer-in
+  // Required
+  google.protobuf.Timestamp transfer_in_exclusivity = 5;
 }
 
-// Records that a contract has been archived, and choices may no longer be exercised on it.
-message ArchivedEvent {
+// Records that a contract has been transferred-in, and it becomes usable on the target domain
+message TransferredInEvent {
 
-  // The ID of this particular event.
+  // The ID of the created contract.
   // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
-  string event_id = 1;
+  string contract_id = 1;
 
-  // The ID of the archived contract.
-  // Must be a valid LedgerString (as described in ``value.proto``).
+  // The ID of the source domain
+  // Must be a valid domain ID
   // Required
-  string contract_id = 2;
+  string source = 2;
 
-  // The template of the archived contract.
+  // The ID of the target domain
+  // Must be a valid domain ID
   // Required
-  Identifier template_id = 3;
+  string target = 3;
 
-  // The parties that are notified of this event. For an ``ArchivedEvent``,
-  // these are the intersection of the stakeholders of the contract in
-  // question and the parties specified in the ``TransactionFilter``. The
-  // stakeholders are the union of the signatories and the observers of
-  // the contract.
-  // Each one of its elements must be a valid PartyIdString (as described
-  // in ``value.proto``).
+  // The ID of the transfer-out event
+  // For correlation capabilities
+  // Must be a valid LedgerString (as described in ``value.proto``).
   // Required
-  repeated string witness_parties = 4;
+  string transfer_out_id = 4;
 }
```

Comment:
@andreaslochbihler-da : is the id of the corresponding transfer-in event available as well? also, can we see who initiated the transfer-in for debugging purposes?